builtin/game/register.lua

```diff
@@ -113,6 +113,8 @@ function core.register_item(name, itemdef)
 	-- Check name
 	if name == nil then
 		error("Unable to register item: Name is nil")
+	elseif name == ":" and itemdef.type ~= "none" then
```

Comment:
When the restriction is instead `itemdef.type == "block"`, mods can define items with the name `:` that override the default hand without it being obvious.

For example,
```lua
minetest.register_item(":", {
    type = "tool"
})
```
causes the default hand to show up with a default "no texture" texture.
![screenshot_20211118_200650](https://user-images.githubusercontent.com/17197562/142522368-20bad382-6ba4-43e2-936d-44d65d801fa7.png)

It makes the most sense to prevent anything else from sharing this name.

Reply:
It might make sense, but it would also break compatibility. Besides, you might want the hand to be a tool in order to display hand wear for instance? I only see a point in disabling overriding the hand with a node as that seems to cause heavy breakage.